--- workers.py	2023-08-30 15:13:45.131604043 +0300
+++ workers.py	2023-08-30 15:19:35.119581079 +0300
@@ -225,7 +225,10 @@
         # The shared secret used for authentication when connecting to the main synapse.
         self.worker_replication_secret = config.get("worker_replication_secret", None)
 
-        self.worker_name = config.get("worker_name", self.worker_app)
+        from os import environ
+        self.worker_name = self.worker_app
+        if self.worker_app:
+             self.worker_name = config.get("worker_name", environ.get("HOSTNAME"))
         self.instance_name = self.worker_name or MAIN_PROCESS_INSTANCE_NAME
 
         # FIXME: Remove this check after a suitable amount of time.
